daemon/src/analyzer.c

```diff
@@ -64,6 +73,10 @@ long long analyzing(const char *path, struct task_details *task)
     long long size = 0;
     DIR *directory = opendir(path);
 
+    double dirs_count = 0;
+    PathStack *stack = (PathStack *)malloc(sizeof(PathStack));
+    initialize_path_stack(stack);
+
     if (directory == NULL)
         return 0;
 
```

Comment:
can this while be a function now ?

Reply:
I guess so, I can try to make it a while if you want.